src/main/java/seedu/moneygowhere/parser/ConsoleParser.java

```diff
@@ -357,106 +265,101 @@ private static ConsoleCommandEditExpense parseCommandEditExpense(String argument
-                    true,
-                    "type"
-            );
-            Option optionOrder = new Option(
-                    "o",
-                    "order",
-                    true,
-                    "Ascending/Descending"
-            );
-            Options options = new Options();
-            options.addOption(optionType);
-            options.addOption(optionOrder);
-            CommandLineParser commandLineParser = new DefaultParser();
-            CommandLine commandline = commandLineParser.parse(options, argumentsArr);
-            String type = commandline.getOptionValue(CONSOLE_COMMAND_SORT_EXPENSE_TYPE);
-            String order = commandline.getOptionValue(CONSOLE_COMMAND_SORT_EXPENSE_ORDER);
-            if (type == null
-                    || !(type.equalsIgnoreCase(CONSOLE_COMMAND_SORT_EXPENSE_TYPE_ALPHABETICAL)
-                    || type.equalsIgnoreCase(CONSOLE_COMMAND_SORT_EXPENSE_TYPE_DATE)
-                    || type.equalsIgnoreCase(CONSOLE_COMMAND_SORT_EXPENSE_TYPE_AMOUNT))) {
+            Options cliOptions = ConsoleParserConfigurations.getCommandSortExpenseOptions();
+            CommandLineParser cliParser = new DefaultParser();
+            CommandLine cli = cliParser.parse(cliOptions, argumentsArr);
+
+            String type = cli.getOptionValue(
+                    ConsoleParserConfigurations.COMMAND_SORT_EXPENSE_ARG_TYPE_LONG
+            );
+            String order = cli.getOptionValue(
+                    ConsoleParserConfigurations.COMMAND_SORT_EXPENSE_ARG_ORDER_LONG
+            );
+
+            /* Checks if mandatory arguments are provided */
+
+            if (type == null || order == null) {
+                throw new ConsoleParserCommandSortExpenseInvalidTypeException();
+            }
+
+            /* Checks if arguments are valid */
+
+            if (type.equalsIgnoreCase(
+                    ConsoleParserConfigurations.COMMAND_SORT_EXPENSE_ARG_TYPE_VAL_ALPHABETICAL)
+                    || type.equalsIgnoreCase(
+                    ConsoleParserConfigurations.COMMAND_SORT_EXPENSE_ARG_TYPE_VAL_AMOUNT)
+                    || type.equalsIgnoreCase(
+                    ConsoleParserConfigurations.COMMAND_SORT_EXPENSE_ARG_TYPE_VAL_DATE)
+            ) {
                 throw new ConsoleParserCommandSortExpenseInvalidTypeException();
             }
-            if (order == null
-                    || !(order.equalsIgnoreCase(CONSOLE_COMMAND_SORT_EXPENSE_ORDER_ASCENDING)
-                    || order.equalsIgnoreCase(CONSOLE_COMMAND_SORT_EXPENSE_ORDER_DESCENDING))) {
+
+            if (order.equalsIgnoreCase(
+                    ConsoleParserConfigurations.COMMAND_SORT_EXPENSE_ARG_ORDER_VAL_ASCENDING)
+                    || order.equalsIgnoreCase(
+                    ConsoleParserConfigurations.COMMAND_SORT_EXPENSE_ARG_ORDER_VAL_DESCENDING)
+            ) {
                 throw new ConsoleParserCommandSortExpenseInvalidTypeException();
             }
```

Comment:
Slight bug here. The `type` and `order` should be !(type.equalsIgnoreCase()) in the if-block before throwing a new ConsoleParserCommandSortExpenseInvalidTypeException()